Flaky python unit test test_publish_compact_and_consume

#### Expected behavior

Seen failure at : https://builds.apache.org/job/pulsar_precommit_cpp/2951/console

```
======================================================================
FAIL: test_publish_compact_and_consume (__main__.PulsarTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "pulsar_test.py", line 557, in test_publish_compact_and_consume
    self.assertEqual(msg0.data(), b'hello-1')
AssertionError: 'hello-0' != 'hello-1'

----------------------------------------------------------------------
```


```
2018-08-06 22:34:41.901 INFO  HandlerBase:53 | [persistent://sample/standalone/ns1/my-python-test_publish_compact_and_consume, my-producer-name] Getting connection from pool
2018-08-06 22:34:42.058 INFO  ProducerImpl:154 | [persistent://sample/standalone/ns1/my-python-test_publish_compact_and_consume, my-producer-name] Created producer on broker [127.0.0.1:55834 -> 127.0.0.1:8885] 
2018-08-06 22:34:42.060 INFO  HandlerBase:53 | [persistent://sample/standalone/ns1/my-python-test_publish_compact_and_consume, my-sub1, 0] Getting connection from pool
2018-08-06 22:34:42.133 INFO  ConsumerImpl:168 | [persistent://sample/standalone/ns1/my-python-test_publish_compact_and_consume, my-sub1, 0] Created consumer on broker [127.0.0.1:55834 -> 127.0.0.1:8885] 
2018-08-06 22:34:42.134 INFO  ConsumerImpl:761 | [persistent://sample/standalone/ns1/my-python-test_publish_compact_and_consume, my-sub1, 0] Closed consumer 0
2018-08-06 22:34:42.135 INFO  HandlerBase:53 | [persistent://sample/standalone/ns1/my-python-test_publish_compact_and_consume, my-sub2, 1] Getting connection from pool
2018-08-06 22:34:42.207 INFO  ConsumerImpl:168 | [persistent://sample/standalone/ns1/my-python-test_publish_compact_and_consume, my-sub2, 1] Created consumer on broker [127.0.0.1:55834 -> 127.0.0.1:8885] 
2018-08-06 22:34:42.265 INFO  ProducerImpl:466 | [persistent://sample/standalone/ns1/my-python-test_publish_compact_and_consume, my-producer-name] Closed producer
Compact still running
{"status":"RUNNING","lastError":""}
Compact still running
{"status":"RUNNING","lastError":""}
Compact Complete now
{"status":"SUCCESS","lastError":""}
2018-08-06 22:34:42.817 INFO  HandlerBase:53 | [persistent://sample/standalone/ns1/my-python-test_publish_compact_and_consume, my-sub1, 2] Getting connection from pool
2018-08-06 22:34:42.818 INFO  ConsumerImpl:168 | [persistent://sample/standalone/ns1/my-python-test_publish_compact_and_consume, my-sub1, 2] Created consumer on broker [127.0.0.1:55834 -> 127.0.0.1:8885] 
```

--- a/pulsar-client-cpp/python/pulsar_test.py
+++ b/pulsar-client-cpp/python/pulsar_test.py
@@ -22,6 +22,7 @@
 from unittest import TestCase, main
 import time
 import os
+import uuid
 from pulsar import Client, MessageId, \
             CompressionType, ConsumerType, PartitionsRoutingMode, \
             AuthenticationTLS, Authentication, AuthenticationToken, InitialPosition
@@ -608,7 +609,7 @@ class PulsarTest(TestCase):
 
     def test_publish_compact_and_consume(self):
         client = Client(self.serviceUrl)
-        topic = 'my-python-test_publish_compact_and_consume'
+        topic = 'compaction_%s' % (uuid.uuid4())
         producer = client.create_producer(topic, producer_name='my-producer-name', batching_enabled=False)
         self.assertEqual(producer.last_sequence_id(), -1)
         consumer = client.subscribe(topic, 'my-sub1', is_read_compacted=True)
@@ -621,7 +622,7 @@ class PulsarTest(TestCase):
         producer.close()
 
         # issue compact command, and wait success
-        url=self.adminUrl + '/admin/v2/persistent/public/default/my-python-test_publish_compact_and_consume/compaction'
+        url='%s/admin/v2/persistent/public/default/%s/compaction' % (self.adminUrl, topic)
         doHttpPut(url, '')
         while True:
             s=doHttpGet(url).decode('utf-8')
@@ -634,6 +635,14 @@ class PulsarTest(TestCase):
                 print("Compact Complete now")
                 print(s)
                 break
+
+        # after compaction completes the compacted ledger is recorded
+        # as a property of a cursor. As persisting the cursor is async
+        # and we don't wait for the acknowledgement of the acknowledgement,
+        # there may be a race if we try to read the compacted ledger immediately.
+        # therefore wait a second to allow the compacted ledger to be updated on
+        # the broker.
+        time.sleep(1.0)
 
         # after compact, consumer with `is_read_compacted=True`, expected read only the second message for same key.
         consumer1 = client.subscribe(topic, 'my-sub1', is_read_compacted=True)

